e2e/stack_test.go

```diff
@@ -83,25 +101,92 @@ func TestE2E(t *testing.T) {
 	})
 	require.NoError(t, err)
 
+	// Run tests concurrently, against the same stack.
+	runningTests := sync.WaitGroup{}
+	runningTests.Add(len(e2eTests))
+
+	for _, test := range e2eTests {
+		t.Run(test.name, func(t *testing.T) {
+			go func() {
+				defer runningTests.Done()
+				test.run(t, stack)
+			}()
+		})
+	}
+
+	runningTests.Wait()
+}
+
+func containsAttributesTx(t *testing.T, stack *e2e.StackConfig) {
+	targetHeight := uint64(5)
+
+	// wait for some blocks to be processed
+	err := stack.WaitL2(int(targetHeight))
+	require.NoError(t, err)
+
+	for i := uint64(2); i < targetHeight; i++ {
```

Comment:
Yeah - I've modified that a couple of times before because it seemed wrong. Block 1 is genesis and has no transactions, so fails the assertion that follows.